lib/matplotlib/tests/test_backend_pdf.py

```diff
@@ -132,3 +133,37 @@ def test_grayscale_alpha():
     ax.imshow(dd, interpolation='none', cmap='gray_r')
     ax.set_xticks([])
     ax.set_yticks([])
+
+
+@cleanup
+def test_pdfpages_accept_pep_519():
+    from tempfile import NamedTemporaryFile
+
+    class FakeFSPathClass(object):
+        def __init__(self, path):
+            self._path = path
+
+        def __fspath__(self):
+            return self._path
+    with NamedTemporaryFile(suffix='.pdf') as tmpfile:
+        with PdfPages(FakeFSPathClass(tmpfile.name)) as pdf:
+            fig, ax = plt.subplots()
+            ax.plot([1, 2], [3, 4])
+            pdf.savefig(fig)
```

Comment:
... but this doesn't: the NamedTemporaryFile still has the file open and savefig writes to it. The other test closes the file and reuses only the name.